Sources/NIOHTTPTypesHTTP1/HTTPTypeConversion.swift

```diff
@@ -0,0 +1,210 @@
+            case "REBIND": self = .REBIND
+            case "UNBIND": self = .UNBIND
+            case "REPORT": self = .REPORT
+            case "UNLINK": self = .UNLINK
+            case "MSEARCH": self = .MSEARCH
+            case "PROPFIND": self = .PROPFIND
+            case "CHECKOUT": self = .CHECKOUT
+            case "PROPPATCH": self = .PROPPATCH
+            case "SUBSCRIBE": self = .SUBSCRIBE
+            case "MKCALENDAR": self = .MKCALENDAR
+            case "MKACTIVITY": self = .MKACTIVITY
+            case "UNSUBSCRIBE": self = .UNSUBSCRIBE
+            case "SOURCE": self = .SOURCE
+            default: self = .RAW(value: rawValue)
+            }
+        }
+    }
+}
+
+extension HTTPRequest.Method {
+    init(_ oldMethod: HTTPMethod) throws {
+        switch oldMethod {
+        case .GET: self = .get
+        case .PUT: self = .put
+        case .ACL: self = .init("ACL")!
+        case .HEAD: self = .head
+        case .POST: self = .post
+        case .COPY: self = .init("COPY")!
+        case .LOCK: self = .init("LOCK")!
+        case .MOVE: self = .init("MOVE")!
+        case .BIND: self = .init("BIND")!
+        case .LINK: self = .init("LINK")!
+        case .PATCH: self = .patch
+        case .TRACE: self = .trace
+        case .MKCOL: self = .init("MKCOL")!
+        case .MERGE: self = .init("MERGE")!
+        case .PURGE: self = .init("PURGE")!
+        case .NOTIFY: self = .init("NOTIFY")!
+        case .SEARCH: self = .init("SEARCH")!
+        case .UNLOCK: self = .init("UNLOCK")!
+        case .REBIND: self = .init("REBIND")!
+        case .UNBIND: self = .init("UNBIND")!
+        case .REPORT: self = .init("REPORT")!
+        case .DELETE: self = .delete
+        case .UNLINK: self = .init("UNLINK")!
+        case .CONNECT: self = .connect
+        case .MSEARCH: self = .init("MSEARCH")!
+        case .OPTIONS: self = .options
+        case .PROPFIND: self = .init("PROPFIND")!
+        case .CHECKOUT: self = .init("CHECKOUT")!
+        case .PROPPATCH: self = .init("PROPPATCH")!
+        case .SUBSCRIBE: self = .init("SUBSCRIBE")!
+        case .MKCALENDAR: self = .init("MKCALENDAR")!
+        case .MKACTIVITY: self = .init("MKACTIVITY")!
+        case .UNSUBSCRIBE: self = .init("UNSUBSCRIBE")!
+        case .SOURCE: self = .init("SOURCE")!
+        case .RAW(value: let value):
+            guard let method = HTTPRequest.Method(value) else {
+                throw HTTP1TypeConversionError.invalidMethod
+            }
+            self = method
+        }
+    }
+}
+
+extension HTTPHeaders {
+    init(_ newFields: HTTPFields) {
+        let fields = newFields.map { ($0.name.rawName, $0.value) }
+        self.init(fields)
+    }
+}
+
+extension HTTPFields {
+    init(_ oldHeaders: HTTPHeaders, splitCookie: Bool) {
+        self.init()
+        self.reserveCapacity(count)
+        for (index, field) in oldHeaders.enumerated() {
+            if index == 0, field.name.lowercased() == "host" {
```

Comment:
RFC7230 has this sentence which is no longer present in RFC9112:

>    Since the Host field-value is critical information for handling a
>    request, a user agent SHOULD generate Host as the first header field
>    following the request-line.

I guess we can't assume that it is always first. HTTPTypes does not preserve the order of pseudo header fields, so we will lose the order of Host (which might be fine?)

Reply:
Fixed in 6da1b1a